app/Svg/Parser.hs

```diff
@@ -71,10 +71,19 @@ performParse :: T.Text -- ^ The title of the graph.
              -> String -- ^ The filename of the file that will be parsed.
              -> SqlPersistM ()
 performParse graphName inputFilename = do
+    deleteExistingGraph graphName
     liftIO . print $ "Parsing graph " ++ T.unpack graphName ++ " from file " ++ inputFilename
     graphFile <- liftIO $ T.readFile (graphPath ++ inputFilename)
     performParseFromMemory graphName graphFile False
 
+-- | Check if a graph exists and delete it if it does
+deleteExistingGraph :: T.Text -> SqlPersistM ()
+deleteExistingGraph graphName = do
+  maybeResponse <- liftIO $ getGraph graphName
```

Comment:
It's good that you found the `getGraph` function, but you shouldn't actually call it here for two reasons. First, it executes a much larger database query than is necessary for this function; and, its type is `IO` rather than `SqlPersistM`, which basically means that `getGraph` executes in its own database transaction, and we should avoid executing multiple database transactions in the same "action" when possible.

Instead, you should take just the part that's relevant for this function (getting the Graph!) and use it here.